src/Symfony/Component/Translation/Catalogue/ReplaceOperation.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Catalogue;
+
+use Symfony\Component\Translation\MetadataAwareInterface;
+
+/**
+ * This will merge and replace all values in $target with values from $source.
+ * It is the equivalent of running array_merge($target, $source). When in conflict,
+ * always take values from $source.
+ *
+ * This operation is metadata aware. It will do the same recursive merge on metadata.
+ *
+ * all = source ∪ target = {x: x ∈ source ∨ x ∈ target}
+ * new = all ∖ target = {x: x ∈ source ∧ x ∉ target}
```

Comment:
I am not sure that we can change the definition of `all` or `new` in a concrete operation. As far as I see it, `all`, `new`, and `obsolete` are defined by the `OperationInterface` and we can only define `result` in a concrete operation.

Reply:
I think you are misstaken. Different concrete operation will result in different "all", "new" and "obsolete". Compare `MergeOperation` with `TargetOperation`.

Reply:
Hm, but in this case we should update the interface's docblock if it specifies a contract we don't comply with.